parser/parse.go

```diff
@@ -105,84 +114,86 @@ func (b *Builder) AddBuildTags(tags ...string) {
 // Get package information from the go/build package. Automatically excludes
 // e.g. test files and files for other platforms-- there is quite a bit of
 // logic of that nature in the build package.
-func (b *Builder) buildPackage(pkgPath string) (*build.Package, error) {
-	// This is a bit of a hack.  The srcDir argument to Import() should
-	// properly be the dir of the file which depends on the package to be
-	// imported, so that vendoring can work properly.  We assume that there is
-	// only one level of vendoring, and that the CWD is inside the GOPATH, so
-	// this should be safe.
-	cwd, err := os.Getwd()
-	if err != nil {
-		return nil, fmt.Errorf("unable to get current directory: %v", err)
+func (b *Builder) importBuildPackage(dir string) (*build.Package, error) {
```

Comment:
I found this coming the other direction because it is improperly creating imports that look like this `v1 "k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1"` .  and that behavior appears to be intentional based on https://golang.org/src/go/build/build.go#L570